gammapy/makers/map.py

```diff
@@ -382,8 +387,11 @@ def run(self, dataset, observation):
             Map dataset.
         """
         kwargs = {"gti": observation.gti}
-        kwargs["meta_table"] = self.make_meta_table(observation)
-        kwargs["meta"] = self._make_metadata(kwargs["meta_table"])
+        if isinstance(observation, Observation):
+            kwargs["meta_table"] = self.make_meta_table(observation)
+            kwargs["meta"] = self._make_metadata(kwargs["meta_table"])
+        else:
+            kwargs["meta"] = observation.meta
```

Comment:
If `observation` is not an Observation, what is it then???

Reply:
a map dataset

Reply:
But, the first argument of the function is a dataset:
> def run(self, dataset, observation):
 So, one can have 2 dataset objects when we call the function?

Reply:
yes

Reply:
So the docstring should be updated!
And even wondering if the name of the argument (`observation`) is now obsolete...